### PR TITLE
task: Update linting pipeline

### PR DESCRIPTION
**Pull request by gabrielmocanu**
Update linting pipeline

### CODE REVIEW

File: .github/workflows/linting.yml [hunk 1]
```diff
@@ -1,29 +1,26 @@
...
-      ################################
-      # Run Linter against code base #
-      ################################
-      - name: Lint Code Base
-        uses: github/super-linter@v4
-        env:
-          VALIDATE_ALL_CODEBASE: false
-          DEFAULT_BRANCH: master
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
```
Comment (teodutu) [inline .github/workflows/linting.yml:26]:
```suggestion
```

File: .github/workflows/linting.yml [hunk 2]
```diff
@@ -1,29 +1,26 @@
...
   pull_request:
     branches:
       - master
 
 jobs:
+  checkpatch:
+    name: Checkpatch
+    runs-on: ubuntu-latest
+    steps:
+      - name: 📥 Checkpatch
```
Comment (teodutu) [inline .github/workflows/linting.yml:13]: Do we really want to use emoticons in scripts (or anywhere else, intermingled with text / code)? Are we boomers already?

File: .github/workflows/linting.yml [hunk 3]
```diff
@@ -1,29 +1,26 @@
...
-    name: Super Linter Runner
+    name: Super Linter
     runs-on: ubuntu-latest
     steps:
-      ##########################
-      # Checkout the code base #
-      ##########################
-      - name: Checkout Code
-        uses: actions/checkout@v3
+      - name: 📥 Super Linter
```
Comment (teodutu) [inline .github/workflows/linting.yml:22]: Ditto :cry:.

**Review by teodutu — CHANGES_REQUESTED** [review]

File: .github/workflows/linting.yml (hunk 2 above)
Comment (gabrielmocanu) [reply .github/workflows/linting.yml:13]: Looks better.

File: .github/workflows/linting.yml (hunk 3 above)
Comment (gabrielmocanu) [reply .github/workflows/linting.yml:22]: Looks better

File: .github/workflows/linting.yml (hunk 1 above)
Comment (gabrielmocanu) [reply .github/workflows/linting.yml:26]: Done.

**Review by teodutu — APPROVED** [review]
Sic transit gloria mundi.

**Pull request merged**